src/node_crypto.cc

```diff
@@ -3156,12 +3210,15 @@ void CipherBase::SetAuthTag(const FunctionCallbackInfo<Value>& args) {
 
   // Restrict GCM tag lengths according to NIST 800-38d, page 9.
   unsigned int tag_len = Buffer::Length(args[0]);
-  if (tag_len > 16 || (tag_len < 12 && tag_len != 8 && tag_len != 4)) {
-    char msg[125];
-    snprintf(msg, sizeof(msg),
-             "Permitting authentication tag lengths of %u bytes is deprecated. "
-             "Valid GCM tag lengths are 4, 8, 12, 13, 14, 15, 16.", tag_len);
-    ProcessEmitDeprecationWarning(cipher->env(), msg, "DEP0090");
+  const int mode = EVP_CIPHER_CTX_mode(cipher->ctx_);
+  if (mode == EVP_CIPH_GCM_MODE) {
+    if (tag_len > 16 || (tag_len < 12 && tag_len != 8 && tag_len != 4)) {
```

Comment:
So, OpenSSL will let through a GCM tag length that's not one of these values but CCM is more strict? So we don't need to do a similar thing for CCM here because OpenSSL does it for us?

Reply:
The GCM authentication tag is simply truncated to the desired length, and OpenSSL permits any such length. In CCM, the authentication tag is part of the algorithm and it is mathematically impossible to use an invalid length: Three bits are used to encode `(t-2)/2` where `t` is the length of the authentication tag and at least one bit must be non-zero. If `t` cannot be encoded, OpenSSL will return an error.